### DISCUSSION TITLE
installing duckdb extensions globally instead of in user homedir

### DISCUSSION THREAD

**Discussion by nbc** [duckdb/duckdb#7807] [General]
Hello,

I run a big rstudio server with many users.

Is there a way to install icu extension for R duckdb package globally to avoid my users the hassle of installing extensions behind a firewall (https://github.com/duckdb/duckdb/issues/3836) ?

I've search everywhere (doc, source code) and didn't find information.

I know how to compile extensions as built-in  but I would like to avoid that too.

Thanks in advance

**Comment by rsund** [comment]
There are some [configuration options](https://duckdb.org/docs/sql/configuration.html)  (e.g. extension_directory and custom_extension_repository) that could help. In addition, you can use path in `INSTALL` so you could download the extensions to be available in your environment. That can be a bit complicated if users have different versions of DuckDB in use, but if you have good practises to install (same versions of) R and R packages, that should work.

I have also had plans to put DuckDB extensions inside of custom R packages to be more easily installable in this kind of scenarios, but due to various reasons that appears to be not so easy unless you just put correct compiled extension inside the package and put the package to custom repository (that you may already have in your environment).

Or if you tailor the DuckDB package a bit you could install the (external) extensions at the same time as DuckDB package as DuckDB [should be able to find extensions also in its own package directory](https://github.com/duckdb/duckdb/pull/3416#discussion_r864963259), but also that can be a hassle with all the versions.
